src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -709,9 +713,9 @@ def run(self):
                     # when the user pressed Control D
                     break
                 else:
+                    self.history.append(text)
                     b_flag, c_flag, outside, cmd = self._special_cases(cmd, outside)
-                    if not self.default_command:
```

Comment:
What is default_command?

Reply:
It's the scope; empty string if not in one.

Reply:
Seems to me then we should still not commit the default command to history?  We wouldn't want empty strings appended.

Reply:
We don't. The default_command is no longer included within history. 
We append "text" to history as long as it is not empty.